src/data/moves/move-attrs/ability-change-attr.ts

```diff
@@ -46,4 +48,18 @@ export class AbilityChangeAttr extends MoveEffectAttr {
       !(this.selfTarget ? user : target).getAbility().hasAttrFlag(AbAttrFlag.UNSUPPRESSABLE_ABILITY)
       && (this.selfTarget ? user : target).getAbility().id !== this.ability;
   }
+
+  /**
+   * If the target is an opponent and is known to have a high-value ability,
+   * grants (+2) effect score.
+   * @see {@linkcode HIGH_VALUE_ABILITIES}
+   */
+  override getEffectScore(user: EnemyPokemon, target: Pokemon, _move: Move): number {
+    const hasHighValueAbility = target
+      .getAbilities({ canApplyOnly: true, revealedOnly: true })
+      .filter((ab) => !ab.passive) // Remove this if passives are made to be overwritten
+      .some((ab) => HIGH_VALUE_ABILITIES.includes(ab.ability.id));
+
+    return hasHighValueAbility && user.isOpponent(target) ? 2 : 0;
```

Comment:
That `+2` number should probably be a constant to be more "descriptive" 

and not a *magic number*